clique/src/service.rs

```diff
@@ -92,6 +94,21 @@ pub trait SpawnHandleT {
     ) -> Pin<Box<dyn Future<Output = Result<(), ()>> + Send>>;
 }
 
+impl<SH: SpawnHandleT> SpawnHandleExt for SH {
+    fn spawn_essential(
+        &self,
+        name: &'static str,
+        task: impl Future<Output = ()> + Send + 'static,
+    ) -> Pin<Box<dyn Future<Output = Result<(), ()>> + Send>> {
+        let (tx, rx) = oneshot::channel();
+        self.spawn(name, async move {
+            task.await;
+            let _ = tx.send(());
+        });
+        Box::pin(rx.map_err(|_| ()))
```

Comment:
nit: I dont like we are loosing an error here - though I know it would probably never be a problem hence nit

Reply:
changed